libs/ledger-live-common/src/families/evm/prepareTransaction.ts

```diff
@@ -0,0 +1,41 @@
+import BigNumber from "bignumber.js";
+import { getFeesEstimation, getGasEstimation } from "../../api/Evm";
+import {
+  EvmTransactionEIP1559,
+  EvmTransactionLegacy,
+  Transaction as EvmTransaction,
+} from "./types";
+import { Account } from "@ledgerhq/types-live";
+
+export const prepareTransaction = async (
+  account: Account,
+  tx: EvmTransaction
+): Promise<EvmTransaction> => {
+  const { currency } = account;
+  const gasLimit = await getGasEstimation(currency, tx);
+  const feeData = await getFeesEstimation(currency);
+
+  // If the blockchain is supporting EIP-1559, use maxFeePerGas & maxPriorityFeePerGas
```

Comment:
Just a suggestion but maybe we could deport it into buildEip1559Transaction and buildLegacyTransaction methods ? It's fine like this tho

Reply:
`buildTransaction` is already a convention in our coin integration model, which I purposely avoided to use because its naming makes it hard to understand what it actually does (creating an unsigned transaction from a transaction). So I guess if we want to split the code, we'll need yet another naming for that to avoid the confusion.

But right here to be honest I don't feel like there enough code or complexity to split it. I'm a little biased though cause I already feel like we are creating mazes of functionalities. So if that's not an issue, I'd prefer to keep it simple here 🙏

Reply:
What I suggest is just using 2 local methods that wouldn't be exported anywhere else, but yeah I agree it's already very simple, it's just that sometimes i prefer using methods over comments ! Feel free to resolve